Topics/Software_Engineering/Deploying_Personal_Website.md

```diff
@@ -0,0 +1,92 @@
+
+If you want to display the same content to every visitor of your site, and you’re looking for a cost-effective solution with lower hosting costs, you might want to opt for a static personal website. However, if you would like to add interactive features like allowing visitors to leave comments on your blog posts that require backend processing, or if you want to experiment with external APIs and databases, a dynamic personal website would be a better choice. 
+
+## Deploying Static Websites
+
+**Prerequisite:** 
+Create a [GitHub](https://github.com/) account
+
+**Step 1: Develop your website**
+
+Locally develop your website on your computer. You can use static site generators like [Hugo](https://gohugo.io/), [Jekyll](https://jekyllrb.com/), [Next.js](https://nextjs.org/), or build a simple website using HTML, CSS, and Javascript. Locally test your website and view it using a development server to visualize your website before deployment.
+
+**Step 2: Create a GitHub repository**
+
+Login to GitHub and [create a new repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository) for your website. [Push](https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/adding-locally-hosted-code-to-github) your files from your local machine to your new GitHub repository. 
+
+**Step 3: Deploy to Netlify**
+
+- Login to [Netlify](https://app.netlify.com/login) with your GitHub account. Authorize Netlify to access your GitHub account.
+
+<img width="1494" alt="Screenshot 2023-11-22 at 11 19 06 AM" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/52506101/4949b73e-3e81-459e-8e6d-7c8b43d197d5">
+
+- Click `Add new site` from the Netlify dashboard. Click `Import an existing project` and select `Deploy with GitHub.` 
+
+<img width="1206" alt="Screenshot 2023-11-22 at 11 20 55 AM" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/52506101/91bab809-3e24-4c11-80d6-fac1006800fd">
+
+- Select the repository that you pushed your website code to. Click `Deploy Site` and Netlify will deploy your site for you. 
+
+After you’ve deployed, Netlify provides a URL, but you can buy a custom domain and use that instead. You may also benefit from setting up continuous deployment so every time you push changes to your GitHub repository, they will be automatically deployed. 
+
+## Deploying Dynamic Websites
+
+**Prerequisites:**
+Create a [GitHub](https://github.com/) account
+
+**Step 1: Develop your website**
+
+Build your dynamic website using a backend framework (ex. Flask, Node.js) and a frontend framework (ex. React) and thoroughly test your website locally using a development server. 
```

Comment:
i've never seen ex. used that way (it might just be me) but maybe replace with e.g.

Reply:
Updated!